test/test-integration/src/test/java/com/alipay/sofa/rpc/server/rest/CustomizeFilterTest.java

```diff
@@ -0,0 +1,157 @@
+
+    private static CustomizeContainerResponseTestFilter customizeContainerResponseTestFilter;
+
+    private static CustomizeClientRequestTestFilter     customizeClientRequestTestFilter;
+
+    private static CustomizeClientResponseTestFilter    customizeClientResponseTestFilter;
+
+    @BeforeClass
+    public static void beforeClass() {
+        customizeContainerRequestTestFilter = new CustomizeContainerRequestTestFilter();
+        customizeContainerResponseTestFilter = new CustomizeContainerResponseTestFilter();
+        customizeClientRequestTestFilter = new CustomizeClientRequestTestFilter();
+        customizeClientResponseTestFilter = new CustomizeClientResponseTestFilter();
+
+        JAXRSProviderManager.registerCustomProviderInstance(customizeContainerRequestTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeContainerResponseTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeClientRequestTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeClientResponseTestFilter);
+
+        providerFilter = new CustomizeTestFilter();
+        List<Filter> providerFilters = new ArrayList<Filter>(2);
+        providerFilters.add(providerFilter);
+
+        ServerConfig restServer = new ServerConfig()
+            .setPort(8583)
+            .setProtocol(RpcConstants.PROTOCOL_TYPE_REST);
+
+        List<ServerConfig> servers = new ArrayList<ServerConfig>(2);
+        servers.add(restServer);
+        providerConfig = new ProviderConfig<RestService>()
+            .setInterfaceId(RestService.class.getName())
+            .setRef(new RestServiceImpl())
+            .setRegister(false)
+            .setServer(servers)
+            .setFilterRef(providerFilters);
+
+        providerConfig.export();
+
+        //rest服务
+        clientFilter = new CustomizeTestFilter();
+        List<Filter> clientFilters = new ArrayList<Filter>(2);
+        clientFilters.add(clientFilter);
+
+        ConsumerConfig<RestService> consumerConfigRest = new ConsumerConfig<RestService>()
+            .setInterfaceId(RestService.class.getName())
+            .setProtocol(RpcConstants.PROTOCOL_TYPE_REST)
+            .setDirectUrl("rest://127.0.0.1:8583")
+            .setTimeout(1000)
+            .setFilterRef(clientFilters)
+            .setApplication(new ApplicationConfig().setAppName("TestClientRest"));
+        filterRestService = consumerConfigRest.refer();
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        JAXRSProviderManager.registerCustomProviderInstance(customizeContainerRequestTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeContainerResponseTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeClientRequestTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeClientResponseTestFilter);
```

Comment:
Need call `removeCustomProviderInstance` after run test case, not `registerCustomProviderInstance`

Reply:
my fault